bigbite/themer
# [BBT-90] Dimensions reusable component

## Description



Completes [BBT-90](https://b5ecom.atlassian.net/browse/BBT-90) - This PR adds a reusable Dimensions component to the plugin. There's a few different formats in which this can be set, so I've tried to cover all of these apart from the global values (`inherit`, `initial` etc.)

The UI has two components:

### Type dropdown

The dropdown contains the 4 different types of format you can use: absolute value (`1rem`), fit content (`fit-content(1rem)`), min content & max content.

### Unit value

The unit value is only usable if you have fit content or absolute value selected as these require a value to be valid.

## Change Log



- Added Dimensions reusable component.
- Added `issCssLengthUnit` to helpers (this may not show as a change as it's also in 2 other PR's)

## Steps to test



- Add some valid dimensions styles to your theme.json. For example:
```json
"core/paragraph": {
  "dimensions": {
    "minHeight": "min-content"
  }
}
```
- Check your value is read currently by the component
- Change some values using the component UI
- Save & export your `theme.json`. Check your changes are saved & exported correctly.

## Screenshots/Videos

### Importing

https://github.com/bigbite/themer/assets/45216035/f02b9bcf-5a20-4ac0-8229-df7ccd064c5d

### Changing values

https://github.com/bigbite/themer/assets/45216035/f7c399a7-76ac-4fe5-ad95-5fa779e8bc4f

### Exporting changes

https://github.com/bigbite/themer/assets/45216035/44220bd5-b07a-41b3-9d3d-590aff58a349

## Checklist:





[BBT-90]: https://b5ecom.atlassian.net/browse/BBT-90?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
